lib/src/sv_sign.c

```diff
@@ -762,6 +762,13 @@ signed_video_set_product_info(signed_video_t *self,
 {
   if (!self) return SV_INVALID_PARAMETER;
 
+  // If ONVIF is available, translate and call ONVIF API
+  if (self->onvif) {
+    onvif_media_signing_vendor_info_t vendor_info;
```

Comment:
Wouldn't `vendor_info` be initialized to all zeros if you set it to `onvif_media_signing_vendor_info_t vendor_info = {0};`

Then you could replace the function with
```
if (firmware_version) {
  strncpy(vendor_info.firmware_version, firmware_version, 255);
}
if (serial_number) {
  strncpy(vendor_info.serial_number, serial_number, 255);
}
if (manufacturer) {
  strncpy(vendor_info.manufacturer, manufacturer, 255);
}
```

We should consider concatenate `manufacturer` and `address` into one string and pass in as `manufacturer` for ONVIF